src/transform/share_normal_form.rs

```diff
@@ -89,7 +92,21 @@ pub fn transform_one(rt: RichTerm) -> RichTerm {
                         if !t.as_ref().is_constant() {
                             let fresh_var = fresh_var();
                             let pos_t = t.pos;
-                            bindings.push((fresh_var.clone(), t));
+                            let field_deps = deps.as_ref().and_then(|deps| deps.stat_fields.get(&id)).cloned().map(Rc::new);
```

Comment:
The same thing is written a second time a bit further, shouldn't it be in a function ?

Reply:
It's slightly different, because this one `stat_fields` while the other uses `dyn_fields`. Let me think 5min if I can write a function generic enough. That being said, you made me realize I forgot to do the `is_non_rec` optimization in the second case, so this duplicated code is not harmless, in the end...